opm/simulators/wells/StandardWell_impl.hpp

```diff
@@ -1065,14 +1065,15 @@ namespace Opm
         if ( bhp_limit_not_defaulted || !this->wellHasTHPConstraints(summaryState) ) {
             // if the BHP limit is not defaulted or the well does not have a THP limit
             // we need to check the BHP limit
-
+            bool operable_under_bhp = false;
             for (int p = 0; p < this->number_of_phases_; ++p) {
-                const double ipr_rate = this->ipr_a_[p] - this->ipr_b_[p] * bhp_limit;
-                if ( (this->isProducer() && ipr_rate < 0.) || (this->isInjector() && ipr_rate > 0.) ) {
-                    this->operability_status_.operable_under_only_bhp_limit = false;
-                    break;
+                double ipr_rate = this->ipr_a_[p] - this->ipr_b_[p] * bhp_limit;
+                if ( (this->isProducer() && ipr_rate >= 0.) || (this->isInjector() && ipr_rate <= 0.) ) {
```

Comment:
If a well is at zero rate, I would expect that it is now more likely to be considered operable, because of numerical noise. Have you considered adding a low but nonzero threshold for this, such as 1e-8 m^3/s? Also, the same threshold could be applied to check that all phases with the wrong flow direction has a very low flow (just noise).

Reply:
I think you point to some important issue here. I will try out your ideas and report back. Currently I think we keep open inoperable wells with the change suggested in this PR. So even thought it fixes an issue where wells are closed too early, we also need to make sure it shuts wells that really are inoperable.

Reply:
I went back to a "sum" approach but this time weighted by the surface density. I.e. to make it more robust in terms of "small" values of gas flowing in the wrong direction. Seems to work fine both for field model and for the MIN_BHP_1 test.

Reply:
That sounds reasonable. So is it exactly the reservoir rate you do then? Or a "close relative"?

Reply:
It is the mass rate. There may be some issues with indexing in the code. I notices that the concept of components and phases are mixed up in the original code. So even though the code is 
`for (int p = 0; p < this->number_of_phases_; ++p) {`
i think it actually loop over the components. i.e. we may mix the water/oil ref density as the code is now. I will need to double check that before this can be undrafted.

Reply:
I agree, it does not matter in terms of results until you add the density. Should preferably have been correct from day 1 of course...